bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/welcome/NAWelcomeHomeHandler.java

```diff
@@ -97,96 +100,164 @@ protected NAWelcomeHome updateReadings() {
+    protected void triggerChannelIfRequired(String channelId) {
+        if(isNewLastEvent) {
+            if(CHANNEL_CAMERA_EVENT.equals(channelId)) {
+                Set<String> detectedObjectTypes = findDetectedObjectTypes(lastEvent);
+                for(String detectedType: detectedObjectTypes) {
+                    triggerChannel(channelId, detectedType);
+                }
+            }
+        }
+        super.triggerChannelIfRequired(channelId);
+    }
+
+    private static Set<String> findDetectedObjectTypes(Optional<NAWelcomeEvent> eventOptional) {
+        Set<String> detectedObjectTypes = new TreeSet<>();
+        if(!eventOptional.isPresent()) {
+            return detectedObjectTypes;
+        }
+
+        NAWelcomeEvent event = eventOptional.get();
+
+        if(event.getPersonId() != null) {
+            detectedObjectTypes.add(CAMERA_EVENT_OPTION_HUMAN_DETECTED);
+        }
+
+        if(NAWebhookCameraEvent.EventTypeEnum.MOVEMENT.toString().equals(event.getType())) {
+            detectedObjectTypes.add(CAMERA_EVENT_OPTION_MOVEMENT_DETECTED);
+        }
+
+        List<NAWelcomeSubEvent> subEvents = event.getEventList();
+        for(NAWelcomeSubEvent subEvent: subEvents) {
+            String detectedObjectType = translateDetectedObjectType(subEvent.getType());
+            detectedObjectTypes.add(detectedObjectType);
+        }
+        return detectedObjectTypes;
+    }
+
+    private static String translateDetectedObjectType(NAWelcomeSubEvent.TypeEnum type) {
+        switch (type) {
+            case HUMAN: return CAMERA_EVENT_OPTION_HUMAN_DETECTED;
+            case ANIMAL: return CAMERA_EVENT_OPTION_ANIMAL_DETECTED;
+            case VEHICLE: return CAMERA_EVENT_OPTION_VEHICLE_DETECTED;
+            default: throw new IllegalArgumentException("Unknown detected object type!");
+        }
+    }
+
+    private Optional<String> findEventMessage() {
+        if(lastEvent.isPresent()) {
+            @Nullable String message = lastEvent.get().getMessage();
+            if (message != null) {
+                return Optional.of(message);
+            }
+
+            return findFirstSubEvent(lastEvent).map(NAWelcomeSubEvent::getMessage);
+        }
+        return Optional.empty();
+    }
+
     /**
      * Returns the Url of the picture
      *
      * @return Url of the picture or null
      */
-    protected String getSnapshotURL(NAWelcomeSnapshot snapshot) {
-        String result = null;
-
-        if (snapshot != null && snapshot.getId() != null && snapshot.getKey() != null) {
-            result = WELCOME_PICTURE_URL + "?" + WELCOME_PICTURE_IMAGEID + "=" + snapshot.getId() + "&"
-                    + WELCOME_PICTURE_KEY + "=" + snapshot.getKey();
-        } else {
-            logger.debug("Unable to build snapshot url for Home : {}", getId());
+    protected Optional<String> findSnapshotURL() {
+        if(lastEvent.isPresent()) {
+            @Nullable NAWelcomeSnapshot snapshot = lastEvent.get().getSnapshot();
```

Comment:
Is nullable required ?

Reply:
yes, otherwise this warning is shown: "Expression 'findFirstSubEvent(lastEvent).map(NAWelcomeSubEvent::getSnapshot).orElse(null)' might evaluate to null but is assigned to a variable that is annotated with @NotNull"